src/messaging/index.ts

```diff
@@ -20,7 +20,7 @@ export function makeTestBoxEvent<K extends keyof TestBoxOutgoingEvents>(
       version: 1,
       sender: MessageSender.PARTNER,
       event,
-      data,
+      data: data || ({} as TestBoxOutgoingEvents[K]),
```

Comment:
🧠 There was a whole body of work to allow for undefined data instead of a blank object. Is that impossible in strict mode for some reason?

Reply:
This was the only way I could get TS to play ball unfortunately. If you want to take a stab tho, please do